contracts/InterchainTokenFactory.sol

```diff
@@ -53,11 +53,11 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
     }
 
     function interchainTokenId(address deployer, bytes32 salt) public view returns (bytes32 tokenId) {
-        tokenId = service.interchainTokenId(address(this), interchainTokenSalt(chainNameHash, deployer, salt));
+        tokenId = service.interchainTokenId(address(0), interchainTokenSalt(chainNameHash, deployer, salt));
```

Comment:
Create a constant for `address(0)`, maybe `TOKEN_FACTORY_DEPLOYER` and add a comment on it?